content/docs/concepts/architecture.md

```diff
@@ -0,0 +1,320 @@
+
+### Composition
+
+Sometimes a Network Service is provided by a graph of {{<color "#008A00" >}}Endpoints{{< /color >}} composed together to serve that workload.
+
+For example, it is likely simplest when providing a Traditional Service Mesh as a Network Service to Compose
+an Envoy Proxy (managed by an Istio or Kuma control plane) with a vL3 (providing a virtual L3 domain):
+
+![Composition](/img/concepts/architecture/composition.svg)
+
+```yaml
+---
+apiVersion: networkservicemesh.io/v1
+kind: NetworkService
+metadata:
+  name: service-mesh@finance.example.com
+spec:
+  payload: IP
+  matches:
+    - source_selector:
+        service: envoy-proxy
+      routes:
+        - destination_selector:
+            service: vl3
+    - source_selector:
+      routes:
+      - destination_selector:
+          service: envoy-proxy
+```
+
+Please note: there is nothing magic about the choice of labels as 'service: ...' as with all labels, the choice is arbitrary,
+its the matching that matters.
+
+#### Selective Composition
+
+Sometimes it is desirable to have different {{<color "#0050EF" >}}Clients{{< /color >}} receive a different composition of {{<color "#008A00" >}}Endpoints{{< /color >}} to provide a Network Service.
+
+For example, imagine that a {{<color "#0050EF" >}}Client{{< /color >}} is version v1.1 of an app foo.  It is known that v1.1 of app foo has a security
+vulnerability.  There is a plan to remediate to foo version v1.2 with the fix.  The schedule for that is six weeks out.
+App foo needs to stay in deployment in the interim.
+
+An expensive IPS can provide protection from the vulnerability.  By keying off of labels provided by the {{<color "#0050EF" >}}Clients{{< /color >}} when
+they Request the Network Service, NSM can selectively interpose an IPS between all instances of foo v1.1 and the vL3
+for the Network Service.
+
+![Selective Composition](/img/concepts/architecture/selective-composition.svg)
+
+```yaml
+---
+apiVersion: networkservicemesh.io/v1
+kind: NetworkService
+metadata:
+  name: secure-vl3@sales.example.com
+spec:
+  payload: IP
+  matches:
+    - source_selector:
+        app: foo
+        version: v1.1
+      routes:
+        - destination_selector:
+            provides: ips
+    - source_selector:
+        performed: ips
+      routes:
+        - destination_selector:
+            provides: vl3
+    - source_selector:
+        routes:
+        - destination_selector:
+            provides: vl3
+```
+
+All other workloads using the Network Service continue normally without the IPS.
+
+Please note: there is nothing magic about the choice of labels as:
+- provides
+- provided
+- app
+- version
+as with all labels, the choice is arbitrary, it's the matching that matters.
+
+### Topologically Aware Endpoint Selection
+
+Topology for both {{<color "#0050EF" >}}Clients{{< /color >}} and {{<color "#008A00" >}}Endpoints{{< /color >}} can be expressed by source or destination labels.  Examples:
+- nodeName
```

Comment:
This list is not list-like in the preview.

Reply:
Fixed.